api/src/main/java/org/openmrs/api/db/hibernate/envers/OpenMRSRevisionEntityListener.java

```diff
@@ -0,0 +1,28 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+
+package org.openmrs.api.db.hibernate.envers;
+
+import org.hibernate.envers.RevisionListener;
+import org.joda.time.DateTime;
+import org.openmrs.api.context.Context;
+
+import java.util.Date;
+
+public class OpenMRSRevisionEntityListener implements RevisionListener {
+	@Override
+	public void newRevision(Object o) {
+		OpenMRSRevisionEntity customRevisionEntity = (OpenMRSRevisionEntity) o;
+		if(Context.getUserContext().isAuthenticated()) {
```

Comment:
Do we have cases of entities being changed without a logged on user?

Reply:
Yes. Without this check we get the below error.

```
ERROR - Context.checkCoreDataset(1164) |2024-05-07T10:18:50,785| Error while setting core global properties
org.springframework.orm.hibernate5.HibernateSystemException: Unable to perform beforeTransactionCompletion callback: null; nested exception is org.hibernate.HibernateException: Unable to perform beforeTransactionCompletion callback: null
	at org.springframework.orm.hibernate5.SessionFactoryUtils.convertHibernateAccessException(SessionFactoryUtils.java:267) ~[spring-orm-5.3.30.jar:5.3.30]
	at org.springframework.orm.hibernate5.HibernateTransactionManager.convertHibernateAccessException(HibernateTransactionManager.java:787) ~[spring-orm-5.3.30.jar:5.3.30]
	at org.springframework.orm.hibernate5.HibernateTransactionManager.doCommit(HibernateTransactionManager.java:653) ~[spring-orm-5.3.30.jar:5.3.30]
	at org.springframework.transaction.support.AbstractPlatformTransactionManager.processCommit(AbstractPlatformTransactionManager.java:743) ~[spring-tx-5.3.30.jar:5.3.30]
	at org.springframework.transaction.support.AbstractPlatformTransactionManager.commit(AbstractPlatformTransactionManager.java:711) ~[spring-tx-5.3.30.jar:5.3.30]
	at org.springframework.transaction.interceptor.TransactionAspectSupport.commitTransactionAfterReturning(TransactionAspectSupport.java:654) ~[spring-tx-5.3.30.jar:5.3.30]
	at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:407) ~[spring-tx-5.3.30.jar:5.3.30]
	at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:119) ~[spring-tx-5.3.30.jar:5.3.30]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.3.30.jar:5.3.30]
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:241) ~[spring-aop-5.3.30.jar:5.3.30]
	at com.sun.proxy.$Proxy176.saveGlobalProperty(Unknown Source) ~[?:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_402]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_402]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_402]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_402]
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:344) ~[spring-aop-5.3.30.jar:5.3.30]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:198) ~[spring-aop-5.3.30.jar:5.3.30]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163) ~[spring-aop-5.3.30.jar:5.3.30]
	at org.springframework.cache.interceptor.CacheInterceptor.lambda$invoke$0(CacheInterceptor.java:54) ~[spring-context-5.3.30.jar:5.3.30]
	at org.springframework.cache.interceptor.CacheAspectSupport.invokeOperation(CacheAspectSupport.java:366) ~[spring-context-5.3.30.jar:5.3.30]
	at org.springframework.cache.interceptor.CacheAspectSupport.execute(CacheAspectSupport.java:420) ~[spring-context-5.3.30.jar:5.3.30]
	at org.springframework.cache.interceptor.CacheAspectSupport.execute(CacheAspectSupport.java:345) ~[spring-context-5.3.30.jar:5.3.30]
	at org.springframework.cache.interceptor.CacheInterceptor.invoke(CacheInterceptor.java:64) ~[spring-context-5.3.30.jar:5.3.30]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.3.30.jar:5.3.30]
	at org.openmrs.aop.LoggingAdvice.invoke(LoggingAdvice.java:123) ~[classes/:?]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.3.30.jar:5.3.30]
	at org.springframework.aop.framework.adapter.MethodBeforeAdviceInterceptor.invoke(MethodBeforeAdviceInterceptor.java:58) ~[spring-aop-5.3.30.jar:5.3.30]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.3.30.jar:5.3.30]
	at org.springframework.aop.framework.adapter.MethodBeforeAdviceInterceptor.invoke(MethodBeforeAdviceInterceptor.java:58) ~[spring-aop-5.3.30.jar:5.3.30]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.3.30.jar:5.3.30]
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:241) ~[spring-aop-5.3.30.jar:5.3.30]
	at com.sun.proxy.$Proxy204.saveGlobalProperty(Unknown Source) ~[?:?]
	at org.openmrs.api.context.Context.checkCoreDataset(Context.java:1149) [classes/:?]
	at org.openmrs.api.context.Context.startup(Context.java:922) [classes/:?]
	at org.openmrs.web.Listener.startOpenmrs(Listener.java:350) [classes/:?]
	at org.openmrs.web.WebDaemon$1.run(WebDaemon.java:42) [classes/:?]
Caused by: org.hibernate.HibernateException: Unable to perform beforeTransactionCompletion callback: null
	at org.hibernate.engine.spi.ActionQueue$BeforeTransactionCompletionProcessQueue.beforeTransactionCompletion(ActionQueue.java:960) ~[hibernate-core-5.6.15.Final.jar:5.6.15.Final]
	at org.hibernate.engine.spi.ActionQueue.beforeTransactionCompletion(ActionQueue.java:525) ~[hibernate-core-5.6.15.Final.jar:5.6.15.Final]
	at org.hibernate.internal.SessionImpl.beforeTransactionCompletion(SessionImpl.java:2439) ~[hibernate-core-5.6.15.Final.jar:5.6.15.Final]
	at org.hibernate.engine.jdbc.internal.JdbcCoordinatorImpl.beforeTransactionCompletion(JdbcCoordinatorImpl.java:449) ~[hibernate-core-5.6.15.Final.jar:5.6.15.Final]
	at org.hibernate.resource.transaction.backend.jdbc.internal.JdbcResourceLocalTransactionCoordinatorImpl.beforeCompletionCallback(JdbcResourceLocalTransactionCoordinatorImpl.java:183) ~[hibernate-core-5.6.15.Final.jar:5.6.15.Final]
	at org.hibernate.resource.transaction.backend.jdbc.internal.JdbcResourceLocalTransactionCoordinatorImpl.access$300(JdbcResourceLocalTransactionCoordinatorImpl.java:40) ~[hibernate-core-5.6.15.Final.jar:5.6.15.Final]
	at org.hibernate.resource.transaction.backend.jdbc.internal.JdbcResourceLocalTransactionCoordinatorImpl$TransactionDriverControlImpl.commit(JdbcResourceLocalTransactionCoordinatorImpl.java:281) ~[hibernate-core-5.6.15.Final.jar:5.6.15.Final]
	at org.hibernate.engine.transaction.internal.TransactionImpl.commit(TransactionImpl.java:101) ~[hibernate-core-5.6.15.Final.jar:5.6.15.Final]
	at org.springframework.orm.hibernate5.HibernateTransactionManager.doCommit(HibernateTransactionManager.java:645) ~[spring-orm-5.3.30.jar:5.3.30]
	... 33 more
Caused by: java.lang.NullPointerException
	at org.openmrs.api.db.hibernate.envers.OpenMRSRevisionEntityListener.newRevision(OpenMRSRevisionEntityListener.java:24) ~[classes/:?]
	at org.hibernate.envers.internal.revisioninfo.DefaultRevisionInfoGenerator.generate(DefaultRevisionInfoGenerator.java:88) ~[hibernate-envers-5.6.15.Final.jar:5.6.15.Final]
	at org.hibernate.envers.internal.synchronization.AuditProcess.getCurrentRevisionData(AuditProcess.java:133) ~[hibernate-envers-5.6.15.Final.jar:5.6.15.Final]
	at org.hibernate.envers.internal.synchronization.AuditProcess.executeInSession(AuditProcess.java:115) ~[hibernate-envers-5.6.15.Final.jar:5.6.15.Final]
	at org.hibernate.envers.internal.synchronization.AuditProcess.doBeforeTransactionCompletion(AuditProcess.java:175) ~[hibernate-envers-5.6.15.Final.jar:5.6.15.Final]
	at org.hibernate.envers.internal.synchronization.AuditProcessManager$1.doBeforeTransactionCompletion(AuditProcessManager.java:47) ~[hibernate-envers-5.6.15.Final.jar:5.6.15.Final]
	at org.hibernate.engine.spi.ActionQueue$BeforeTransactionCompletionProcessQueue.beforeTransactionCompletion(ActionQueue.java:954) ~[hibernate-core-5.6.15.Final.jar:5.6.15.Final]
	at org.hibernate.engine.spi.ActionQueue.beforeTransactionCompletion(ActionQueue.java:525) ~[hibernate-core-5.6.15.Final.jar:5.6.15.Final]
	at org.hibernate.internal.SessionImpl.beforeTransactionCompletion(SessionImpl.java:2439) ~[hibernate-core-5.6.15.Final.jar:5.6.15.Final]
	at org.hibernate.engine.jdbc.internal.JdbcCoordinatorImpl.beforeTransactionCompletion(JdbcCoordinatorImpl.java:449) ~[hibernate-core-5.6.15.Final.jar:5.6.15.Final]
	at org.hibernate.resource.transaction.backend.jdbc.internal.JdbcResourceLocalTransactionCoordinatorImpl.beforeCompletionCallback(JdbcResourceLocalTransactionCoordinatorImpl.java:183) ~[hibernate-core-5.6.15.Final.jar:5.6.15.Final]
	at org.hibernate.resource.transaction.backend.jdbc.internal.JdbcResourceLocalTransactionCoordinatorImpl.access$300(JdbcResourceLocalTransactionCoordinatorImpl.java:40) ~[hibernate-core-5.6.15.Final.jar:5.6.15.Final]
	at org.hibernate.resource.transaction.backend.jdbc.internal.JdbcResourceLocalTransactionCoordinatorImpl$TransactionDriverControlImpl.commit(JdbcResourceLocalTransactionCoordinatorImpl.java:281) ~[hibernate-core-5.6.15.Final.jar:5.6.15.Final]
	at org.hibernate.engine.transaction.internal.TransactionImpl.commit(TransactionImpl.java:101) ~[hibernate-core-5.6.15.Final.jar:5.6.15.Final]
	at org.springframework.orm.hibernate5.HibernateTransactionManager.doCommit(HibernateTransactionManager.java:645) ~[spring-orm-5.3.30.jar:5.3.30]
	... 33 more
```

Reply:
Can we use something like "Anonymous User" in such instances?

Reply:
Can you try track the source of such edits? They probably should not happen in the very first place. Don't you think so?

Reply:
The description of the global property `registrationcore.patientNameSearch` gets changed during the startup. Here is a screenshot of the global_property_AUD table.

![Screenshot from 2024-05-11 20-53-56](https://github.com/openmrs/openmrs-core/assets/61653572/bf5dfc48-df9b-405a-8857-67354fef6a16)

Reply:
Can we defer changing that to after one has logged in?

Reply:
29 changes happen before the user logs in. Would we be able to address all these changes?

Reply:
Do you have a list of these changes?

Reply:
Here are the changes that happen before the login.

**global_property**

```
#####Querying table: global_property_AUD#####
layout.address.format 61 1 <org.openmrs.layout.address.AddressTemplate>
     <nameMappings class="properties">
       <property name="postalCode" value="Location.postalCode"/>
       <property name="address2" value="Location.address2"/>
       <property name="address1" value="Location.address1"/>
       <property name="country" value="Location.country"/>
       <property name="stateProvince" value="Location.stateProvince"/>
       <property name="cityVillage" value="Location.cityVillage"/>
     </nameMappings>
     <sizeMappings class="properties">
       <property name="postalCode" value="10"/>
       <property name="address2" value="40"/>
       <property name="address1" value="40"/>
       <property name="country" value="10"/>
       <property name="stateProvince" value="10"/>
       <property name="cityVillage" value="10"/>
     </sizeMappings>
     <lineByLineFormat>
       <string>address1</string>
       <string>address2</string>
       <string>cityVillage stateProvince country postalCode</string>
     </lineByLineFormat>
   </org.openmrs.layout.address.AddressTemplate> XML description of address formats null null null null null null null null null 61 1715840908701 null 2024-05-16 11:58:28 

registrationcore.patientNameSearch 62 1 registrationcore.BasicPatientNameSearch Specifies a bean used for the auto suggest name feature. null null null null null null null null null 62 1715840914694 null 2024-05-16 11:58:34 

layout.address.format 83 1 <org.openmrs.layout.address.AddressTemplate>
     <nameMappings class="properties">
       <property name="postalCode" value="Location.postalCode"/>
       <property name="address2" value="Location.address2"/>
       <property name="address1" value="Location.address1"/>
       <property name="country" value="Location.country"/>
       <property name="stateProvince" value="Location.stateProvince"/>
       <property name="cityVillage" value="Location.cityVillage"/>
     </nameMappings>
     <sizeMappings class="properties">
       <property name="postalCode" value="10"/>
       <property name="address2" value="40"/>
       <property name="address1" value="40"/>
       <property name="country" value="10"/>
       <property name="stateProvince" value="10"/>
       <property name="cityVillage" value="10"/>
     </sizeMappings>
     <lineByLineFormat>
       <string>address1</string>
       <string>address2</string>
       <string>cityVillage stateProvince country postalCode</string>
     </lineByLineFormat>
   </org.openmrs.layout.address.AddressTemplate> null null null null null null null null null null 83 1715840974570 null 2024-05-16 11:59:34 

registrationcore.patientNameSearch 88 1 registrationcore.ExistingPatientNameSearch null null null null null null null null null null 88 1715840976319 null 2024-05-16 11:59:36 

registrationcore.patientNameSearch 89 1 registrationcore.BasicPatientNameSearch null null null null null null null null null null 89 1715840976494 null 2024-05-16 11:59:36 

```
**role**

```
#####Querying table: role_AUD#####
Application: Administers System 63 1 63 1715840973308 null 2024-05-16 11:59:33 

Application: Configures Appointment Scheduling 63 1 63 1715840973308 null 2024-05-16 11:59:33 

Application: Configures Forms 63 1 63 1715840973308 null 2024-05-16 11:59:33 

Application: Configures Metadata 63 1 63 1715840973308 null 2024-05-16 11:59:33 

Application: Edits Existing Encounters 63 1 63 1715840973308 null 2024-05-16 11:59:33 

Application: Enters ADT Events 63 1 63 1715840973308 null 2024-05-16 11:59:33 

Application: Enters Vitals 63 1 63 1715840973308 null 2024-05-16 11:59:33 

Application: Has Super User Privileges 63 1 63 1715840973308 null 2024-05-16 11:59:33 

Application: Manages Atlas 63 1 63 1715840973308 null 2024-05-16 11:59:33 

Application: Manages Provider Schedules 63 1 63 1715840973308 null 2024-05-16 11:59:33 

Application: Records Allergies 63 1 63 1715840973308 null 2024-05-16 11:59:33 

Application: Registers Patients 63 1 63 1715840973308 null 2024-05-16 11:59:33 

Application: Requests Appointments 63 1 63 1715840973308 null 2024-05-16 11:59:33 

Application: Schedules And Overbooks Appointments 63 1 63 1715840973308 null 2024-05-16 11:59:33 

Application: Schedules Appointments 63 1 63 1715840973308 null 2024-05-16 11:59:33 

Application: Sees Appointment Schedule 63 1 63 1715840973308 null 2024-05-16 11:59:33 

Application: Uses Capture Vitals App 63 1 63 1715840973308 null 2024-05-16 11:59:33 

Application: Uses Patient Summary 63 1 63 1715840973308 null 2024-05-16 11:59:33 

Application: Writes Clinical Notes 63 1 63 1715840973308 null 2024-05-16 11:59:33 

Application: Administers System 64 1 64 1715840973950 null 2024-05-16 11:59:33 

Application: Configures Appointment Scheduling 65 1 65 1715840973987 null 2024-05-16 11:59:33 

Application: Configures Forms 66 1 66 1715840974025 null 2024-05-16 11:59:34 

Application: Configures Metadata 67 1 67 1715840974070 null 2024-05-16 11:59:34 

Application: Edits Existing Encounters 68 1 68 1715840974101 null 2024-05-16 11:59:34 

Application: Enters ADT Events 69 1 69 1715840974129 null 2024-05-16 11:59:34 

Application: Enters Vitals 70 1 70 1715840974157 null 2024-05-16 11:59:34 

Application: Has Super User Privileges 71 1 71 1715840974191 null 2024-05-16 11:59:34 

Application: Manages Atlas 72 1 72 1715840974216 null 2024-05-16 11:59:34 

Application: Manages Provider Schedules 73 1 73 1715840974240 null 2024-05-16 11:59:34 

Application: Records Allergies 74 1 74 1715840974273 null 2024-05-16 11:59:34 

Application: Registers Patients 75 1 75 1715840974300 null 2024-05-16 11:59:34 

Application: Requests Appointments 76 1 76 1715840974326 null 2024-05-16 11:59:34 

Application: Schedules And Overbooks Appointments 77 1 77 1715840974357 null 2024-05-16 11:59:34 

Application: Schedules Appointments 78 1 78 1715840974388 null 2024-05-16 11:59:34 

Application: Sees Appointment Schedule 79 1 79 1715840974409 null 2024-05-16 11:59:34 

Application: Uses Capture Vitals App 80 1 80 1715840974433 null 2024-05-16 11:59:34 

Application: Uses Patient Summary 81 1 81 1715840974462 null 2024-05-16 11:59:34 

Application: Writes Clinical Notes 82 1 82 1715840974495 null 2024-05-16 11:59:34 
```
**role_role**

```
#####Querying table: role_role_AUD#####
63 Application: Administers System Privilege Level: High 2 63 1715840973308 null 2024-05-16 11:59:33 

63 Application: Configures Appointment Scheduling Privilege Level: High 2 63 1715840973308 null 2024-05-16 11:59:33 

63 Application: Configures Forms Privilege Level: High 2 63 1715840973308 null 2024-05-16 11:59:33 

63 Application: Configures Metadata Privilege Level: High 2 63 1715840973308 null 2024-05-16 11:59:33 

63 Application: Edits Existing Encounters Privilege Level: High 2 63 1715840973308 null 2024-05-16 11:59:33 

63 Application: Enters ADT Events Privilege Level: High 2 63 1715840973308 null 2024-05-16 11:59:33 

63 Application: Enters Vitals Privilege Level: High 2 63 1715840973308 null 2024-05-16 11:59:33 

63 Application: Has Super User Privileges Privilege Level: High 2 63 1715840973308 null 2024-05-16 11:59:33 

63 Application: Manages Atlas Privilege Level: High 2 63 1715840973308 null 2024-05-16 11:59:33 

63 Application: Manages Provider Schedules Privilege Level: High 2 63 1715840973308 null 2024-05-16 11:59:33 

63 Application: Records Allergies Privilege Level: High 2 63 1715840973308 null 2024-05-16 11:59:33 

63 Application: Registers Patients Privilege Level: High 2 63 1715840973308 null 2024-05-16 11:59:33 

63 Application: Requests Appointments Privilege Level: High 2 63 1715840973308 null 2024-05-16 11:59:33 

63 Application: Schedules And Overbooks Appointments Privilege Level: High 2 63 1715840973308 null 2024-05-16 11:59:33 

63 Application: Schedules Appointments Privilege Level: High 2 63 1715840973308 null 2024-05-16 11:59:33 

63 Application: Sees Appointment Schedule Privilege Level: High 2 63 1715840973308 null 2024-05-16 11:59:33 

63 Application: Uses Capture Vitals App Privilege Level: High 2 63 1715840973308 null 2024-05-16 11:59:33 

63 Application: Uses Patient Summary Privilege Level: High 2 63 1715840973308 null 2024-05-16 11:59:33 

63 Application: Writes Clinical Notes Privilege Level: High 2 63 1715840973308 null 2024-05-16 11:59:33 

64 Application: Administers System Privilege Level: High 0 64 1715840973950 null 2024-05-16 11:59:33 

65 Application: Configures Appointment Scheduling Privilege Level: High 0 65 1715840973987 null 2024-05-16 11:59:33 

66 Application: Configures Forms Privilege Level: High 0 66 1715840974025 null 2024-05-16 11:59:34 

67 Application: Configures Metadata Privilege Level: High 0 67 1715840974070 null 2024-05-16 11:59:34 

68 Application: Edits Existing Encounters Privilege Level: High 0 68 1715840974101 null 2024-05-16 11:59:34 

69 Application: Enters ADT Events Privilege Level: High 0 69 1715840974129 null 2024-05-16 11:59:34 

70 Application: Enters Vitals Privilege Level: High 0 70 1715840974157 null 2024-05-16 11:59:34 

71 Application: Has Super User Privileges Privilege Level: High 0 71 1715840974191 null 2024-05-16 11:59:34 

72 Application: Manages Atlas Privilege Level: High 0 72 1715840974216 null 2024-05-16 11:59:34 

73 Application: Manages Provider Schedules Privilege Level: High 0 73 1715840974240 null 2024-05-16 11:59:34 

74 Application: Records Allergies Privilege Level: High 0 74 1715840974273 null 2024-05-16 11:59:34 

75 Application: Registers Patients Privilege Level: High 0 75 1715840974300 null 2024-05-16 11:59:34 

76 Application: Requests Appointments Privilege Level: High 0 76 1715840974326 null 2024-05-16 11:59:34 

77 Application: Schedules And Overbooks Appointments Privilege Level: High 0 77 1715840974357 null 2024-05-16 11:59:34 

78 Application: Schedules Appointments Privilege Level: High 0 78 1715840974388 null 2024-05-16 11:59:34 

79 Application: Sees Appointment Schedule Privilege Level: High 0 79 1715840974409 null 2024-05-16 11:59:34 

80 Application: Uses Capture Vitals App Privilege Level: High 0 80 1715840974433 null 2024-05-16 11:59:34 

81 Application: Uses Patient Summary Privilege Level: High 0 81 1715840974462 null 2024-05-16 11:59:34 

82 Application: Writes Clinical Notes Privilege Level: High 0 82 1715840974495 null 2024-05-16 11:59:34
```
**users**
```
#####Querying table: users_AUD#####
3 84 1 2024-05-16 11:59:34 2024-05-15 20:11:57 null null null 0 3-4 clerk 2 2 3 null 84 1715840974706 null 2024-05-16 11:59:34 

4 85 1 2024-05-16 11:59:34 2024-05-15 20:11:57 null null null 0 4-2 nurse 2 2 4 null 85 1715840974754 null 2024-05-16 11:59:34 

5 86 1 2024-05-16 11:59:34 2024-05-15 20:11:57 null null null 0 5-9 doctor 2 2 5 null 86 1715840974800 null 2024-05-16 11:59:34 

6 87 1 2024-05-16 11:59:34 2024-05-15 20:11:57 null null null 0 6-7 sysadmin 2 2 6 null 87 1715840974848 null 2024-05-16 11:59:34 

```

Reply:
@wikumChamith can we log those changes as done under the daemon system account?

Reply:
@dkayiwa I updated the PR.

Reply:
Do we now have all tables logged?

Reply:
If we could get [this](https://github.com/openmrs/openmrs-core/pull/4631) merged we could enable auditing to the all the tables except user properties: https://github.com/openmrs/openmrs-core/pull/4631

Reply:
Can we also look into why user properties is not logged?

Reply:
It is caused by this: https://github.com/openmrs/openmrs-core/pull/4620#discussion_r1595464149

I still haven't found a way to mitigate this issue other than manually creating the audit tables.

Reply:
Can you try one last time?

Reply:
sure, thanks for the resource.